hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -131,8 +133,12 @@ private boolean shouldReduceSendingRate(MessageSendingResult result) {
         return !result.isRetryLater() && subscriptionAllowsResending(result);
     }
 
+    private boolean messageSentSucceeded(MessageSendingResult result) {
+        return result.succeeded() || (result.isClientError() && !subscription.getSerialSubscriptionPolicy().isRetryClientErrors());
+    }
+
     private boolean subscriptionAllowsResending(MessageSendingResult result) {
```

Comment:
i dont quite get the naming here: so subscription allows resending if sending failed?